### PR TITLE
Monitor failed job runs

### PR DESCRIPTION
**Pull request by ljwoodley**
Possible solution to issue #100?

Note, this only sends an email alert with the attached log on job failure. I figured since successful runs are logged in the db we wouldn't need to create a log file for that.

### CODE REVIEW

File: etl/run_etl.R [hunk 1]
```diff
@@ -0,0 +1,36 @@
+library(redcapcustodian)
+library(dotenv)
+library(callr)
+library(argparse)
+
+set_script_run_time()
+
+parser <- ArgumentParser()
+parser$add_argument("script_name", nargs=1, help="Script to be run")
```
Comment (pbchase) [inline etl/run_etl.R:9]: Could you allow for optional parameters? it's quite rare but we do pass an argument to one of out ETLs in the rcc.ctsit repo

File: etl/run_etl.R [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+} else {
+  script_name <- "etl/dummy_script.R"
+  if(!fs::file_exists(script_name)) {
+    stop(sprintf("Specified file, %s, does not exist", script_name))
+  }
+}
+
+tryCatch({
+  rscript(script_name, stderr = "log.txt")
```
Comment (pbchase) [inline etl/run_etl.R:25]: ...and then use the arguments here?

**Review by pbchase — CHANGES_REQUESTED** [review]
This is super simple and it _works_. We could use it as-is, but I'd like to see two more features. Could you add these?

1. Add the ability to pass optional arguments to an etl.
2. implement the same functionality in the existing [render_report.R](https://github.com/ctsit/redcapcustodian/blob/master/report/render_report.R) and its copy in the [study_template](https://github.com/ctsit/redcapcustodian/tree/master/study_template/report)

**Comment by pbchase** [comment]
Here's some example code I wrote to test a different way of specifying the optional arguments to the script called by `run_etl.R`

```r
library(argparse)

parser <- ArgumentParser()

parser$add_argument("file", nargs = 1, help = "RScript to be run")
parser$add_argument("optional_args", nargs='+', help = "optional arguments to the script to be run")

args <- parser$parse_args()

paste0("The first argument, file, has a value of '", args$file, "'")
paste0("The optional arguments that follow are '", paste(args$optional_args, collapse = " "), "'")
```

Could you revise `run_etl.R` to use use this model for optional arguments?

**Comment by ljwoodley** [comment]
> Here's some example code I wrote to test a different way of specifying the optional arguments to the script called by `run_etl.R`
> 
> ```r
> library(argparse)
> 
> parser <- ArgumentParser()
> 
> parser$add_argument("file", nargs = 1, help = "RScript to be run")
> parser$add_argument("optional_args", nargs='+', help = "optional arguments to the script to be run")
> 
> args <- parser$parse_args()
> 
> paste0("The first argument, file, has a value of '", args$file, "'")
> paste0("The optional arguments that follow are '", paste(args$optional_args, collapse = " "), "'")
> ```
> 
> Could you revise `run_etl.R` to use use this model for optional arguments?

I don't understand this change?

`parser$add_argument("optional_args", nargs='+', help = "optional arguments to the script to be run")` does not allow for an optional argumnet as it requires at least one value when the script runs due to `nargs='+`'.

`Rscript etl/run_etl.R etl/test_failure_alert.R` fails with this error
 
Screenshot 2024-10-14 at 5 16 06 PM


Also, `Rscript etl/run_etl.R etl/test_failure_alert.R arg1 arg2` is not parsing the arguments properly.

Screenshot 2024-10-14 at 5 17 07 PM

@pbchase

**Review by pbchase — CHANGES_REQUESTED** [review]
I made some suggestions.

We'll make changes to `render_report.R` later after the rcc.grace PR is finished.

**Review by pbchase — APPROVED** [review]

**Pull request merged**